cmake/compiler.cmake

```diff
@@ -211,6 +211,13 @@ option(ENABLE_WERROR "Make all compiler warnings into errors" OFF)
 
 option(ENABLE_UB_SANITIZER "Make the compiler generate runtime code to perform undefined behaviour checks" OFF)
 
+option(ENABLE_HARDENING "Enable compiler options that harden against memory corruption attacks" ON)
+if (ENABLE_HARDENING)
+    set(HARDENING_FLAGS "-fstack-protector")
```

Comment:
Shouldn't we consider `LDFLAGS='-Wl,-z,relro -Wl,-z,bindnow'` within this issue or as a follow up? (They're described on the [Debian's hardening page](https://wiki.debian.org/Hardening).)

Reply:
Done